src/lib/pipeline/m2/index.js

```diff
@@ -8,39 +9,75 @@ class M2 extends THREE.Group {
 
   static cache = {};
 
-  constructor(path, data, skinData) {
+  constructor(path, data, skinData, instanceOpts) {
     super();
 
+    const instance = instanceOpts || null;
```

Comment:
I haven't used defaults for function arguments in JS before. Sounds like a good idea, though.